### PR TITLE
BugFix: quantization tools will throw a ValueError in HistogramCollector

### PR DESCRIPTION
**Pull request by xusworld**
Fix bug and spelling errors.

### CODE REVIEW

**Comment by xusworld** [comment]
@Fengyi-Li

File: onnxruntime/python/tools/quantization/calibrate.py [hunk 1]
```diff
@@ -491,7 +490,7 @@ def collect_for_entropy(self, name_to_arr):
                 old_histogram = self.histogram_dict[tensor]
                 self.histogram_dict[tensor] = self.merge_histogram(old_histogram, data_arr, min_value, max_value, threshold)
             else:
-                hist, hist_edges = np.histogram(data_arr, self.num_quantized_bins, range=(-threshold, threshold))
+                hist, hist_edges = np.histogram(data_arr, 8192, range=(-threshold, threshold))
```
Comment (chilo-ms) [inline onnxruntime/python/tools/quantization/calibrate.py:493]: Please don't hardcode using 8192.
We are going to make 'num_bins' and 'num_quantized_bins' be configurable, by doing so you can set num_bins=8192 and num_quantized_bin=255 or other values that can achieve better accuracy for your model. Please expect our OR in a few days.

File: onnxruntime/python/tools/quantization/calibrate.py [hunk 2]
```diff
@@ -566,19 +565,19 @@ def get_entropy_threshold(self, histogram, num_quantized_bins):
...
         num_half_quantized_bin = num_quantized_bins // 2
         
-        kl_divergence = np.zeros(zero_bin_index - num_half_quantized_bin + 1)
-        thresholds = [(0, 0) for i in range(kl_divergence.size)] 
+        kl_divergence = np.zeros((zero_bin_index - num_half_quantized_bin + 1) // num_half_quantized_bin + 1)
+        thresholds = [(0, 0) for i in range(kl_divergence.size)]
 
-        for i in range(num_half_quantized_bin, zero_bin_index + 1, 1):
-            start_index = zero_bin_index - i 
+        for i in range(num_half_quantized_bin, zero_bin_index + 1, num_half_quantized_bin):
```
Comment (chilo-ms) [inline onnxruntime/python/tools/quantization/calibrate.py:576]: Could you elaborate more why you want to change here? or what document are you referencing to? 
The naive way is to increase the size of candidate distribution/bin by 1 every time until it reaches the limit.

**Comment by chilo-ms** [comment]
We recently added a PR https://github.com/microsoft/onnxruntime/pull/10204 to make 'num_bins' and 'num_quantized_bins' be configurable.

**Comment by xusworld** [comment]
@chilo-ms Hi Chilo，cause you have updated your quantization implementation, I create a new pr #10261 and will create another one to fix quantization bugs in your code, maybe we can do it together. This pr will close soon.

Email : dongjiaxus@163.com
WeChat: GolangMaster

**Pull request closed without merging**